atom/browser/net/atom_url_loader_factory.cc

```diff
@@ -215,8 +237,9 @@ void AtomURLLoaderFactory::StartLoadingBuffer(
     return;
   }
 
-  SendContents(std::move(client), ToResponseHead(dict),
-               node::Buffer::Data(buffer), node::Buffer::Length(buffer));
+  SendContents(
+      std::move(client), ToResponseHead(dict),
+      std::string(node::Buffer::Data(buffer), node::Buffer::Length(buffer)));
```

Comment:
Is it okay to convert this into `std::string` and transmit ? Can we reuse `WriteDataToProducerHandle` and provide the buffer data to mojo pipe ?

Reply:
Using `std::string` to store binary data is a safe pattern and commonly used in Chromium, its network classes like `DataURLLoaderFactory` are also using `std::string` to store binary data.

The `StringDataPipeProducer` writes the data asynchronously and needs the original data to be available until the write is done, copying the buffer to a string is the easiest implementation. If we use `WriteDataToProducerHandle` directly we have to handle the case when one write can not write all data. 

Our old `URLRequestBufferJob` implementation copies the data too so there is no performance loss, it is possible to save the copy completely with a few hacks, and we can do it in future.